### PR TITLE
Docker howto

### PR DESCRIPTION
**Pull request by hushchyn-mikhail**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,6 +1,16 @@
 FROM olantwin/ship-base:20180627-online
 
-ADD . /FairShip
+# Copy FairShip scripts
+COPY . /FairShip
```
Comment (olantwin) [inline Dockerfile]: Why do you replace the `ADD` here? What are the advantages of `COPY`?

File: Dockerfile [hunk 2]
```diff
@@ -1,6 +1,16 @@
...
 
-RUN aliBuild -c shipdist/ --defaults fairship build FairShip --no-local ROOT \
-	&& aliBuild clean --aggressive-cleanup
```
Comment (olantwin) [inline Dockerfile]: Why remove the cleanup? I prefer keeping the image size as small as possible.

File: Dockerfile [hunk 3]
```diff
@@ -1,6 +1,16 @@
...
+# Build FairShip
+RUN aliBuild -c shipdist/ --defaults fairship build FairShip --no-local ROOT
+
+# Additional library for OpenGL
+RUN yum install -y mesa-dri-drivers
+
+# Fix problems with fonts
+RUN yum install -y dejavu-lgc-sans-fonts
```
Comment (olantwin) [inline Dockerfile]: I think adding the packages should happen in the base image to minimise build times: <https://github.com/olantwin/ship-base>

File: Dockerfile (hunk 1 above)
Comment (hushchyn-mikhail) [reply Dockerfile]: This is nothing more than a habit, and I found it in tips https://nickjanetakis.com/blog/docker-tip-2-the-difference-between-copy-and-add-in-a-dockerile.

File: Dockerfile (hunk 3 above)
Comment (hushchyn-mikhail) [reply Dockerfile]: Agree. 

Ship-base Dockerfile is great. Can you add this link into the FairShip Dockerfile or ship-base on dockerhub? It is very useful.

File: Dockerfile (hunk 2 above)
Comment (hushchyn-mikhail) [reply Dockerfile]: Agree. I'll put it back. I removed it for tests, but now we can put it back.

**Comment by hushchyn-mikhail** [comment]
About the ```ENTRYPOINT```. Yes, we can warn users about it. But now I can not image how can it lead to a problem.

**Comment by olantwin** [comment]
> About the `ENTRYPOINT`. Yes, we can warn users about it. But now I can not image how can it lead to a problem.

There might be some cases where users already call `alienv` in their scripts, which could cause trouble (but should not fail silently in that case, so should be easy to fix).

File: Dockerfile (hunk 3 above)
Comment (olantwin) [reply Dockerfile]: I hope we can also move it to ShipSoft eventually (@FonsRademakers ?)

I'll create a pull-request to your repo adding the info. (and should probably add some more info on Dockerhub as well)

File: Dockerfile (hunk 1 above)
Comment (olantwin) [reply Dockerfile]: This is a good point. This is indeed a good change!

File: Dockerfile (hunk 2 above)
Comment (olantwin) [reply Dockerfile]: Thanks!

File: Dockerfile (hunk 3 above)
Comment (olantwin) [reply Dockerfile]: Also a good time to maybe rebuild ship-base...

**Pull request merged**